src/main/java/org/broadinstitute/consent/http/service/EmailService.java

```diff
@@ -0,0 +1,446 @@
+                .distinct()
+                .toList();
+        for (User user : distinctUsers) {
+            Writer template = templateHelper.getNewDARRequestTemplate(SERVER_URL, user.getDisplayName(), collection.getDarCode());
+            Map<String, String> data = retrieveForNewDAR(collection.getDarCode(), user);
+            Optional<Response> response = sendGridAPI.sendNewDARRequests(user.getEmail(), data.get("entityId"), data.get("electionType"), template);
+            saveEmailAndResponse(
+                response.orElse(null),
+                collection.getDarCode(),
+                null,
+                user.getUserId(),
+                EmailType.NEW_DAR,
+                template
+            );
+        }
+    }
+
+    public void sendCollectMessage(Integer electionId) throws MessagingException, IOException, TemplateException {
+        Set<User> chairs = userDAO.findUsersForElectionsByRoles(
+                Collections.singletonList(electionId),
+                Collections.singletonList(UserRoles.CHAIRPERSON.getRoleName()));
+        for (User chair : chairs) {
+            Map<String, String> data = retrieveForCollect(electionId, chair);
+            String collectUrl = generateCollectVoteUrl(SERVER_URL, data.get("electionType"), data.get("entityId"), data.get("electionId"));
+            Writer template = templateHelper.getCollectTemplate(data.get("userName"), data.get("electionType"), data.get("entityName"), collectUrl);
+            Optional<Response> response = sendGridAPI.sendCollectMessage(data.get("email"), data.get("entityName"), data.get("electionType"), template);
+            saveEmailAndResponse(
+                response.orElse(null),
+                data.get("electionId"),
+                null,
+                Integer.valueOf(data.get("dacUserId")),
+                EmailType.COLLECT,
+                template
+            );
+        }
+    }
+
+    public void sendReminderMessage(Integer voteId) throws IOException, TemplateException {
+        Map<String, String> data = retrieveForVote(voteId);
+        String voteUrl = generateUserVoteUrl(SERVER_URL, data.get("electionType"), data.get("voteId"), data.get("entityId"), data.get("rpVoteId"));
+        Writer template = templateHelper.getReminderTemplate(data.get("userName"), data.get("electionType"), data.get("entityName"), voteUrl);
+        Optional<Response> response = sendGridAPI.sendReminderMessage(data.get("email"), data.get("entityName"), data.get("electionType"), template);
+        voteDAO.updateVoteReminderFlag(voteId, true);
+        saveEmailAndResponse(
+            response.orElse(null),
+            data.get("electionId"),
+            voteId,
+            Integer.valueOf(data.get("dacUserId")),
+            EmailType.REMINDER,
+            template
+        );
+    }
+
+    public void sendDarNewCollectionElectionMessage(List<User> users, DarCollection darCollection) throws IOException, TemplateException {
```

Comment:
Question about `email_preferences`: When should it be taken into account? In this function it takes a list of users as an argument. With its current usage in `DarCollectionService.java ln:824`, that list of users is obtained by a query that does not take `email_preferences` into account.

In contrast, `sendClosedDataSetElectionsMessage` queries users within the function itself with a DAO method that does take `email_preferences` into account.

Reply:
IIRC, that preference is an admin only feature. I think we should honor it regardless, so I'll dig into this detail and see if we can treat it more consistently here as opposed to elsewhere. Looks like there are some conflicts too - another reason to dig back in.